src/kbmod/work_unit.py

```diff
@@ -541,6 +548,237 @@ def to_fits(self, filename, overwrite=False):
 
         hdul.writeto(filename, overwrite=overwrite)
 
+    def to_fits_shard(self, filename, directory, overwrite=False):
```

Comment:
As a minor naming note, I would suggest `to_sharded_fits`. But I'm fine if you want to keep it as-is.